src/nix/repl.cc

```diff
@@ -624,8 +624,17 @@ void NixRepl::loadFlake(const std::string & flakeRefS)
                 .useRegistries = !evalSettings.pureEval,
                 .allowMutable  = !evalSettings.pureEval,
             }),
-        v);
-    addAttrsToScope(v);
+        *v);
+
+    auto f = v->attrs->get(state->symbols.create(fragment));
```

Comment:
I think this won’t work properly if `fragment` has more than one level (`foo.bar`). You probably need to split it first with `parseAttrPath` and recursively get the right value.

Reply:
I'll give that a shot. While we are at it, is there a function or method somewhere used to traverse the multiple outputs to find a match?